### PR TITLE
Tigris: simplify `ObjectID` and filter usage

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Ref #900.

In this PR:
- New package `internal/handlers/tigris/tigrisdb/filter` to replace Tigris' filter package which is based on generics and doesn't fit our custom types.
- The path `internal/handlers/tigris/tigrisdb` is chosen this way as it is supposed to be similar to `internal/handlers/pg/pgdb`.
- `tjson.ObjectID` function is deleted as it's not needed.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [x] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers, Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/tigris/object_id_smoke_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/FerretDB/FerretDB/integration"
+	"github.com/FerretDB/FerretDB/integration/setup"
+)
+
+// TODO This is a temporary test to check how string identifiers would work.
```
Comment (rumyantseva) [inline integration/tigris/object_id_smoke_test.go:30]: @AlekSi should I delete those smoke tests and write compat tests in this task? Or should I create a separate ticket for them?

**Review by w84thesun — COMMENTED** [review]
Overall looks good. Just a small question.

Did I get it correct that we will use strings as `_id` values to insert them into Tigris?

**Comment by rumyantseva** [comment]
@w84thesun sort of :) From Tigris' perspective it's [type: string, format: byte](https://docs.tigrisdata.com/overview/schema#string) to store binary data. From our perspective, it's still `[12]byte`.

**Review by w84thesun — DISMISSED** [review]

File: internal/handlers/tigris/msg_update.go [hunk 2]
```diff
@@ -196,9 +196,9 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 
 // update updates documents by _id.
 func (h *Handler) update(ctx context.Context, sp fetchParam, doc *types.Document) (int, error) {
-	id := must.NotFail(doc.Get("_id")).(types.ObjectID)
-	f := must.NotFail(filter.Eq("_id", tjson.ObjectID(id)).Build())
-	h.L.Sugar().Debugf("Filter: %s", f)
+	id := must.NotFail(doc.Get("_id"))
+	f := must.NotFail(filter.Eq("_id", id).Build())
```
Comment (AlekSi) [inline internal/handlers/tigris/msg_update.go:200]: I really think that having a separate package with three files, types, functions, builders, etc is too much for a single line: `json.Marshal(map[string]any{"_id": id})`

Let's keep it for now, but I think we should really simplify it when we will work on query pushdown

**Review by AlekSi — APPROVED** [review]

File: integration/tigris/object_id_smoke_test.go (hunk 1 above)
Comment (AlekSi) [reply integration/tigris/object_id_smoke_test.go:30]: Let's move them the `intergation` package

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Review by w84thesun — DISMISSED** [review]

**Review by AlekSi — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Pull request merged**